### PR TITLE
fix: Improved sidebar's UX and styling 

### PR DESCRIPTION
**Pull request by isabensusan**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Relates to #2437
Relates to #2416

## Mobile & Desktop Screenshots/Recordings
Before:
image

After:
image

+ Animations
Rewatch GIF Recording - 2024-01-25 at 7 19 35 PM

## Steps to QA
N/A

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x ] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

image

### CODE REVIEW

File: styles/globals.css [hunk 1]
```diff
@@ -22,23 +22,64 @@ h5,
...
   @apply border;
   @apply border-transparent;
 }
 
-input[type="checkbox"]:focus-visible;
-[role="checkbox"]:focus-visible;
-[role="radio"]:focus-visible;
-[role="combobox"]:focus-visible;
-[role="switch"]:focus-visible;
+input[type="checkbox"]:focus-visible,
```
[the remark this answers is not in the document]
Comment (nickytonline) [reply styles/globals.css:30]: I put up a separate PR for this as it's not related to your styling changes. See #2537.

File: components/shared/AppSidebar/InsightsPanel.tsx [hunk 2]
```diff
@@ -27,26 +37,30 @@ export const InsightsPanel = ({ title, username, insights, type, isLoading }: In
...
+                  className={"w-4 h-4 ml-auto transition-all" + (open ? " rotate-180" : "")}
+                  title={`close ${title} panel`}
+                />
               </ClientOnly>
             </h3>
           </Thumb>
         </Root>
       </Collapsible.Trigger>
-      <Collapsible.Content>
+      <Collapsible.Content className="CollapsibleContent">
```
Comment (nickytonline) [inline components/shared/AppSidebar/InsightsPanel.tsx:53]: As discussed in our call, prefer kebab-case for naming.

File: styles/globals.css [hunk 3]
```diff
@@ -22,23 +22,64 @@ h5,
...
+.transition-all {
+  transition-property: all;
+  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+  transition-duration: 0.15s;
+}
+
+/* collapsible component animation */
+.CollapsibleContent {
+  overflow: hidden;
+}
```
Comment (nickytonline) [inline styles/globals.css:57]: refactor (blocking): use the Tailwind CSS class, `overflow-hidden` for this instead.

File: styles/globals.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (isabensusan) [reply styles/globals.css:30]: ok cool!

**Pull request merged**